### PR TITLE
[Keyboard] Add yandrstudio/buff67v3 keyboard.

### PR DESCRIPTION
**Pull request by jiaxin96**
## Description
add buff67v3 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yandrstudio/buff67v3/buff67v3.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const rgblight_segment_t PROGMEM my_capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 1, HSV_RED}
+);
+
+const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    my_capslock_layer
+);
+
+bool led_update_kb(led_t led_state) {
+    rgblight_set_layer_state(0, led_state.caps_lock);
```
Comment (drashna) [inline keyboards/yandrstudio/buff67v3/buff67v3.c:30]:
```suggestion
    if (led_update_user(led_state) {
        rgblight_set_layer_state(0, led_state.caps_lock);
    }
```

File: keyboards/yandrstudio/buff67v3/buff67v3.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    rgblight_layers = my_rgb_layers;
+}
+
+#endif
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    switch(keycode) {
+        case LOCK_GUI:
+            process_magic(GUI_TOG, record);
+            return false;
```
Comment (drashna) [inline keyboards/yandrstudio/buff67v3/buff67v3.c:45]: Why?  

Why not just use GUI_TOG.

File: keyboards/yandrstudio/buff67v3/config.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+#ifdef RGBLIGHT_ENABLE
+
+#    define RGB_DI_PIN B4
+#    define RGBLED_NUM 1
+#    define DRIVER_LED_TOTAL RGBLED_NUM
+#    define RGBLIGHT_SLEEP
+#    define RGBLIGHT_VAL_STEP 10
+#    define RGBLIGHT_LIMIT_VAL 200
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED true
```
Comment (drashna) [inline keyboards/yandrstudio/buff67v3/config.h:49]: This is only for RGB Matrix.```suggestion
```

File: keyboards/yandrstudio/buff67v3/config.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/yandrstudio/buff67v3/config.h:50]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/yandrstudio/buff67v3/readme.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# buff67v3
+
+![buff67v3](https://i.imgur.com/idPbNDL.jpg)
```
Comment (drashna) [inline keyboards/yandrstudio/buff67v3/readme.md:3]:
```suggestion
![buff67v3](https://i.imgur.com/idPbNDLh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/yandrstudio/buff67v3/buff67v3.c (hunk 2 above)
Comment (drashna) [reply keyboards/yandrstudio/buff67v3/buff67v3.c:48]: To be blunt, "how it shows up on via" isn't a good reason for this implementation, ESPECIALLY at the keyboard level.   You are potentially breaking expected behavior for anyone that uses the board, and wants to use F23.  

If you absolutely must implement this, then you need to move this implementation to the via keymap. And there you can use USER00-USER15 to use a custom keycode in the via json.

Comment (jiaxin96) [reply keyboards/yandrstudio/buff67v3/buff67v3.c:48]: thanks，i have used USER00. It's really troublesome that via's documentation didn't mention this.

File: keyboards/yandrstudio/buff67v3/buff67v3.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    if (led_update_user(led_state)) {
+        rgblight_set_layer_state(0, led_state.caps_lock);
+    }
+    return true;
+}
+
+void keyboard_post_init_kb(void) {
+    rgblight_reload_from_eeprom();
+    rgblight_layers = my_rgb_layers;
+}
```
Comment (drashna) [inline keyboards/yandrstudio/buff67v3/buff67v3.c:39]:
```suggestion
    keyboard_post_init_user();
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by keyboard-magpie — COMMENTED** [review]
As mentioned on your other PR, this needs an info.json too

**Review by keyboard-magpie — CHANGES_REQUESTED** [review]
As above

File: keyboards/yandrstudio/buff67v3/info.json [hunk 7]
```diff
@@ -0,0 +1,361 @@
+{
+    "keyboard_name": "buff67v3",
+    "processor": "STM32F103",
+    "usb": {
+        "pid": "0xAA88",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/yandrstudio/buff67v3/info.json:7]:
```suggestion
    "keyboard_name": "buff67v3",
    "manufacturer": "BUFF_Y&R",
    "processor": "STM32F103",
    "bootloader": "stm32duino",
    "usb": {
        "vid": "0xAA96",
        "pid": "0xAA88",
        "device_version": "1.0.0"
    },
```

File: keyboards/yandrstudio/buff67v3/config.h [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xAA96
+#define PRODUCT_ID      0xAA88
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    BUFF_Y&R
+#define PRODUCT         buff67v3
+
```
Comment (fauxpark) [inline keyboards/yandrstudio/buff67v3/config.h:26]:
```suggestion
```

File: keyboards/yandrstudio/buff67v3/rules.mk [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = STM32F103
+
+# Bootloader selection
+BOOTLOADER = stm32duino
+
```
Comment (fauxpark) [inline keyboards/yandrstudio/buff67v3/rules.mk:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/yandrstudio/buff67v3/info.json [hunk 10]
```diff
@@ -0,0 +1,364 @@
+{
+    "keyboard_name": "buff67v3",
+    "manufacturer": "BUFF_Y&R",
+    "processor": "STM32F103",
+    "bootloader": "stm32duino",
+    "usb": {
+        "vid": "0xAA96",
```
Comment (fauxpark) [inline keyboards/yandrstudio/buff67v3/info.json:7]:
```suggestion
    "processor": "STM32F103",
    "bootloader": "stm32duino",
    "usb": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**